internal/extsvc/github/pulls.go

```diff
@@ -752,16 +798,72 @@ func (c *Client) GetOpenPullRequestByRefs(ctx context.Context, owner, name, base
 		return nil, fmt.Errorf("expected 1 pull request, got %d instead", len(results.Repository.PullRequests.Nodes))
 	}
 
-	pr := results.Repository.PullRequests.Nodes[0].PullRequest
-	pr.Participants = results.Repository.PullRequests.Nodes[0].Participants.Nodes
-	pr.TimelineItems = results.Repository.PullRequests.Nodes[0].TimelineItems.Nodes
+	node := results.Repository.PullRequests.Nodes[0]
+	pr := node.PullRequest
+	pr.Participants = node.Participants.Nodes
+	pr.TimelineItems = node.TimelineItems.Nodes
+
+	if node.TimelineItems.PageInfo.HasNextPage {
+		if err := c.loadRemainingTimelineItems(ctx, &pr, node.TimelineItems.PageInfo.EndCursor); err != nil {
+			return nil, err
+		}
+	}
 
 	return &pr, nil
 }
 
+func (c *Client) loadRemainingTimelineItems(ctx context.Context, pr *PullRequest, afterCursor string) error {
```

Comment:
I think I'd prefer this API:

```suggestion
func (c *Client) loadTimelineItems(ctx context.Context, id string, afterCursor string) ([]TimelineItem, error) {
// [...]
}
```

or maybe even:

```suggestion
func (c *Client) loadTimelineItems(ctx context.Context, id string, pageInfo PageInfo) ([]TimelineItem, error) {
  for pageInfo.HasNextPage {
    // [...]
   }
}
```

Then you can save yourself the `if` in the caller.

The most important bit is that I don't think passing in the `*PullRequest` to be mutated is worth it. But this is not a blocker.

Reply:
I changed it to a function that takes the page info and returns the exhaustive list of additional timeline items. Making it a `loadTimelineItems` doesn't work though, because 95% of pull requests don't need additional items and hence we want the standard case to remain only 1 request, I think.

Reply:
Ah, that's fine! The rename was unintentional, sorry about that.